File: 8 kyu/Reversed Strings.js
/*
https://www.codewars.com/kata/5168bb5dfe9a00b126000018

Complete the solution so that it reverses the string passed into it.

'world'  =>  'dlrow'
'word'   =>  'drow'
*/


// my solution
function solution(str){
    const b = [];
    str.split('').forEach(shifter = (x) => b.unshift(x));
    return b.join('');
}

console.log(solution("world"));  // dlrow


// other solutions below. apparently I have overthought about it again lol

//! alternative sol-1
function solution(str){
    return str.split('').reverse().join('');  
}

//! alternative sol-2
const solution = str => str.split('').reverse().join('');

//! alternative sol-3
const solution = s => [...s].reverse().join('')

File: Java/8kyu/Reversed Strings.java
/*
https://www.codewars.com/kata/5168bb5dfe9a00b126000018

Complete the solution so that it reverses the string passed into it.

'world'  =>  'dlrow'
'word'   =>  'drow'
*/


// my solution
import java.util.stream.IntStream;

public class Kata {
  public static String solution(String str) {
    StringBuilder res = new StringBuilder();
    
    for (int i = str.length() - 1; i >= 0; i--) {
      res.append(str.charAt(i));
    }
    
    return res.toString();
  }
}

//! alternative-solution
public class Kata {
    public static String solution(String str) {
      return new StringBuilder(str).reverse().toString();
    }
}